README.md

```diff
@@ -325,6 +326,191 @@ const commonProps = {
 
 A warning is displayed when no fallback image is passed. This warning can be disabled in special circumstances. To disable this warning, look in the [warnings section](#warnings).
 
+#### ImgixProvider Component
+
+> New 🪄 ✨
+
+The [`<ImgixProvider>`]() Higher Order Component (HOC), makes its [props](#props) available to any nested `<Imgix>` component in your React application.
+
+For example, by rendering `<ImgixProvider>` at the top level of your application with `imgixParams` defined, all your `<Imgix>` components will have access to the same `imgixParams`.
+
+```jsx
+import React from "react";
+import Imgix, { ImgixProvider } from "react-imgix";
+import HomePage from "./components/HomePage";
+
+function App() {
+  return (
+    <div className="App">
+      <header className="App-header">
+        <ImgixProvider imgixParams={{ ar: "16:9" }}>
+          <div className="intro-blurb">{/* ... */}</div>
+          <div className="gallery">
+            <Imgix src="https://assets.imgix.net/examples/pione.jpg" />
+            <Imgix src="https://sdk-test.imgix.net/ساندویچ.jpg" />
+          </div>
+        </ImgixProvider>
+      </header>
+    </div>
+  );
+}
+
+export default App;
+```
+
+So that the generated HTML looks something like
+
+```html
+<div class="gallery">
+  <img
+    src="https://assets.imgix.net/examples/pione.jpg?auto=format&ar=16%3A9&"
+    ...
+  />
+  <img
+    src="https://sdk-test.imgix.net/%D8%B3%D8%A7%D9%86%D8%AF%D9%88%DB%8C%DA%86.jpg?auto=format&ar=16%3A9&"
+    ...
+  />
+</div>
+```
+
+You can take advantage of this behavior to use 2-step, or partial, URLs with the `<Imgix>` component. By defining the [`domain`](#domain--string-optional) prop on the Provider, it can be made accessible to all nested `<Imgix>` components.
```

Comment:
I'm fine with referring to partial paths as "2-step" internally but I'm wondering if it could be potentially confusing for our users to have two terms refer to the same thing? This is based off of past experiences with terms like DPR and fixed-sized srcsets.

My 2¢ is to stick to partial or full path URLs only as it's more intuitive.